app/uk/gov/hmrc/cataloguefrontend/leakdetection/LeakDetectionService.scala

```diff
@@ -107,17 +133,21 @@ class LeakDetectionService @Inject() (
               .sortBy(_.lineNumber)
               .sortBy(_.filePath)
           )
-      }
-      .toSeq
-      .sortBy(_.leaks.length)
-      .reverse
-      .sortBy(_.priority)
-    )
 
-  def reportWarnings(reportId: String)(implicit hc: HeaderCarrier): Future[Seq[LeakDetectionWarning]] =
-    leakDetectionConnector.leakDetectionWarnings(reportId)
+  private def filterSummaries(summaries: Seq[LeakDetectionRepositorySummary], includeWarnings: Boolean, includeExemptions: Boolean, includeViolations: Boolean): Seq[LeakDetectionRepositorySummary] = {
```

Comment:
This looks a bit fiddly! Would something like this work?

```scala
  private def filterSummaries(summaries: Seq[LeakDetectionRepositorySummary], includeWarnings: Boolean, includeExemptions: Boolean, includeViolations: Boolean): Seq[LeakDetectionRepositorySummary] = {
    def filter(s: LeakDetectionRepositorySummary) =
      (includeWarnings && (s.warningCount > 0)) || (includeExemptions && (s.excludedCount > 0)) || (includeViolations && (s.unresolvedCount > 0))

    summaries.filter(filter)
  }
```

Maybe it's just essential complexity though.